### PR TITLE
LT-1611: Case flags ui

### PR DESCRIPTION
**Pull request by a-bek**
API changes: https://github.com/uktrade/lite-api/pull/164

Changes:

1.  Display Flags in frontend
2. 4 HTTP requests to `lite-api` squashed to 1.
3. lite-accordian for expanding case rows.

### CODE REVIEW

File: ui_automation_tests/step_defs/test_organisation.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 def verify_registered_organisation(driver, context):
     if not context.org_registered_status:
         driver.find_element_by_id("show-filters-link").click()
+        driver.find_element_by_id("filter-box").click()
```
Comment (andytaylorkainos) [inline ui_automation_tests/step_defs/test_organisation.py]: single quotations please

File: ui_automation_tests/step_defs/test_assign_users_to_queue.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def user_is_on_case_list(driver, internal_info):
 
 @then('user is not assignee on case list')
 def user_is_not_on_case_list(driver, context):
-    assert "No users assigned" in CaseListPage(driver).get_text_of_assignees(context.case_id), "No users assigned text is not displayed"
+    assert "No users assigned" in CaseListPage(driver).get_text_of_assignees(driver, context.case_id), "No users assigned text is not displayed"
```
Comment (andytaylorkainos) [inline ui_automation_tests/step_defs/test_assign_users_to_queue.py]: single quotations please

File: ui_automation_tests/step_defs/test_assign_case_flags_to_case.py [hunk 3]
```diff
@@ -100,3 +101,22 @@ def add_a_flag(driver, add_uae_flag):
...
+    elements = Shared(driver).get_rows_in_lite_table()
+    no = utils.get_element_index_by_text(elements, context.case_id)
+    assert "(3 of " in elements[no].text
+
+
+@then("I dont see 3 out of text")
+def dont_see_three_out_of(driver, context):
+    elements = Shared(driver).get_rows_in_lite_table()
+    no = utils.get_element_index_by_text(elements, context.case_id)
+    assert "(3 of " not in elements[no].text
```
Comment (andytaylorkainos) [inline ui_automation_tests/step_defs/test_assign_case_flags_to_case.py]: single quotations please

File: ui_automation_tests/step_defs/test_assign_case_flags_to_case.py [hunk 4]
```diff
@@ -100,3 +101,22 @@ def add_a_flag(driver, add_uae_flag):
...
+@when('I add a flag called New at level Case')
+def add_a_suspicious_flag(driver, add_new_flag):
+    pass
+
+
+@then("I see 3 out of text")
+def three_out_of_text(driver, context):
+    elements = Shared(driver).get_rows_in_lite_table()
+    no = utils.get_element_index_by_text(elements, context.case_id)
+    assert "(3 of " in elements[no].text
```
Comment (andytaylorkainos) [inline ui_automation_tests/step_defs/test_assign_case_flags_to_case.py]: single quotations please

File: ui_automation_tests/fixtures/add_a_flag.py [hunk 5]
```diff
@@ -28,7 +28,19 @@ def add_suspicious_flag(driver, context):
     assert context.flag_name in Shared(driver).get_text_of_table()
 
 
-@fixture(scope="session")
+@fixture(scope="function")
+def add_new_flag(driver, context):
+    flags_page = FlagsPages(driver)
+    flags_page.click_add_a_flag_button()
+    extra_string = str(utils.get_formatted_date_time_d_h_m_s())
+    context.flag_name = "New" + extra_string
```
Comment (andytaylorkainos) [inline ui_automation_tests/fixtures/add_a_flag.py]: single quotations please

File: ui_automation_tests/conftest.py [hunk 6]
```diff
@@ -262,3 +269,41 @@ def no_permissions(driver):
...
+def go_to_edit_flags(driver):
+    OrganisationPage(driver).click_edit_organisation_flags()
+
+
+@then('the previously created organisations flag is assigned') # noqa
+def assert_flag_is_assigned(driver, context):
+    assert OrganisationPage(driver).is_organisation_flag_applied(context.flag_name)
+
+
+@when("I click chevron") # noqa
```
Comment (andytaylorkainos) [inline ui_automation_tests/conftest.py]: single quotations please

File: ui_automation_tests/fixtures/add_a_flag.py [hunk 7]
```diff
@@ -4,7 +4,7 @@
 from pages.shared import Shared
 
 
-@fixture(scope="session")
+@fixture(scope="function")
```
Comment (andytaylorkainos) [inline ui_automation_tests/fixtures/add_a_flag.py]: single quotations please

File: ui_automation_tests/fixtures/add_a_flag.py [hunk 8]
```diff
@@ -16,7 +16,7 @@ def add_uae_flag(driver, context):
     assert context.flag_name in Shared(driver).get_text_of_table()
 
 
-@fixture(scope="session")
+@fixture(scope="function")
```
Comment (andytaylorkainos) [inline ui_automation_tests/fixtures/add_a_flag.py]: single quotations please

File: ui_automation_tests/fixtures/add_a_flag.py [hunk 9]
```diff
@@ -28,7 +28,19 @@ def add_suspicious_flag(driver, context):
...
```
Comment (andytaylorkainos) [inline ui_automation_tests/fixtures/add_a_flag.py]: single quotations please

File: ui_automation_tests/fixtures/add_a_flag.py [hunk 10]
```diff
@@ -28,7 +28,19 @@ def add_suspicious_flag(driver, context):
...
+    flags_page.click_add_a_flag_button()
+    extra_string = str(utils.get_formatted_date_time_d_h_m_s())
+    context.flag_name = "New" + extra_string
+    flags_page.enter_flag_name(context.flag_name)
+    flags_page.select_flag_level("Case")
+    Shared(driver).click_submit()
+    assert context.flag_name in Shared(driver).get_text_of_table()
+
+
+@fixture(scope="function")
```
Comment (andytaylorkainos) [inline ui_automation_tests/fixtures/add_a_flag.py]: single quotations please

File: ui_automation_tests/conftest.py [hunk 11]
```diff
@@ -262,3 +269,41 @@ def no_permissions(driver):
 @given("I create report summary picklist") # noqa
 def add_report_summary_picklist(add_a_report_summary_picklist):
     pass
+
+
+@then("I see the added flags on the queue") # noqa
```
Comment (andytaylorkainos) [inline ui_automation_tests/conftest.py]: single quotations please

File: ui_automation_tests/pages/queues_pages.py [hunk 12]
```diff
@@ -18,14 +18,14 @@ def click_add_a_queue_button(self):
 
     def is_case_on_the_list(self, app_id):
         self.driver.set_timeout_to(0)
-        no = len(self.driver.find_elements_by_link_text(app_id))
+        no = len(self.driver.find_elements_by_css_selector("[href*='" + app_id + "']"))
```
Comment (andytaylorkainos) [inline ui_automation_tests/pages/queues_pages.py]: single quotations please (this one is mean, i know)

File: ui_automation_tests/step_defs/test_assign_case_flags_to_case.py [hunk 13]
```diff
@@ -100,3 +101,22 @@ def add_a_flag(driver, add_uae_flag):
...
 def add_a_suspicious_flag(driver, add_suspicious_flag):
     pass
+
+
+@when('I add a flag called New at level Case')
+def add_a_suspicious_flag(driver, add_new_flag):
+    pass
+
+
+@then("I see 3 out of text")
```
Comment (andytaylorkainos) [inline ui_automation_tests/step_defs/test_assign_case_flags_to_case.py]: single quotations please

File: ui_automation_tests/step_defs/test_assign_case_flags_to_case.py [hunk 14]
```diff
@@ -100,3 +101,22 @@ def add_a_flag(driver, add_uae_flag):
...
+def three_out_of_text(driver, context):
+    elements = Shared(driver).get_rows_in_lite_table()
+    no = utils.get_element_index_by_text(elements, context.case_id)
+    assert "(3 of " in elements[no].text
+
+
+@then("I dont see 3 out of text")
```
Comment (andytaylorkainos) [inline ui_automation_tests/step_defs/test_assign_case_flags_to_case.py]: single quotations please

File: ui_automation_tests/step_defs/test_assign_users_to_queue.py [hunk 15]
```diff
@@ -26,7 +26,7 @@ def assign_user_to_case(driver, internal_info, context):
 
 @then('user is assignee on case list')
 def user_is_on_case_list(driver, context):
-    assert context.user_name in CaseListPage(driver).get_text_of_assignees(context.case_id), "user name " + context.user_name + " is not an assignee on case list"
+    assert context.user_name in CaseListPage(driver).get_text_of_assignees(driver, context.case_id), "user name " + context.user_name + " is not an assignee on case list"
```
Comment (andytaylorkainos) [inline ui_automation_tests/step_defs/test_assign_users_to_queue.py]: single quotations please

File: ui_automation_tests/pages/queues_pages.py [hunk 16]
```diff
@@ -18,15 +18,20 @@ def click_add_a_queue_button(self):
...
         page_number = 1
         while no == 0:
             page_number += 1
-            next_page = url + '&page=' + str(page_number)
-            self.driver.get(next_page)
-            no = len(self.driver.find_elements_by_link_text(app_id))
+            next_page = url
+
+            if 'queue_id' not in url:
+                next_page = url + '?queue_id=00000000-0000-0000-0000-000000000001'
```
[the remark this answers is not in the document]
Comment (efahy) [reply ui_automation_tests/pages/queues_pages.py]: Actually, this is test code - you can ignore this comment if you like

File: ui_automation_tests/fixtures/add_a_flag.py (hunk 10 above)
Comment (andytaylorkainos) [reply ui_automation_tests/fixtures/add_a_flag.py]: you missed this one :)

File: assets/javascripts/cases.js [hunk 17]
```diff
@@ -0,0 +1,5 @@
+function openTableFold(element) {
+	$(element).parent().parent().find(".app-expanded-row__item, .app-expanded-row__item--invert").toggle();
+	$(element).parent().parent().toggleClass("open");
+	return false;
```
[the remark this answers is not in the document]
Comment (a-bek) [reply assets/javascripts/cases.js]: The problem with expanding everything by default when JS is disabled is that the size of the page can blow up. It's supposed to be a search view of cases providing some metadata, it's harder to search when some rows are huge, so I would argue not to expand when JS is disabled - if the user wants to see all information for a case, they just open it.

[the remark this answers is not in the document]
Comment (efahy) [reply assets/javascripts/cases.js]: That sounds reasonable

File: ui_automation_tests/step_defs/test_organisation.py (hunk 1 above)
Comment (andytaylorkainos) [reply ui_automation_tests/step_defs/test_organisation.py]: please dont ignore this :-)

File: ui_automation_tests/step_defs/test_assign_users_to_queue.py (hunk 15 above)
Comment (andytaylorkainos) [reply ui_automation_tests/step_defs/test_assign_users_to_queue.py]: dont ignore this

File: ui_automation_tests/step_defs/test_assign_users_to_queue.py (hunk 2 above)
Comment (andytaylorkainos) [reply ui_automation_tests/step_defs/test_assign_users_to_queue.py]: pls

**Pull request merged**